color and db fix

MrBlaTi opened the pull request:
@nbatsukh please check if jobs are still working properly

MrBlaTi committed:
multiple fixes
fixed db metadata cleaning
fixed db panic
added db test
fixed overlapping use problem when executing a script on one planet multiple times
fixed problem with multiple planets using one tmpfile to use in tableformatter

MrBlaTi committed:
removed temporary file

MrBlaTi committed:
added ski tmp and chronJob directories,
tmpfiles of tableformatter and execDBCommand are now stored in tmp

MrBlaTi committed:
colors

MrBlaTi committed:
multiple fixes
fixed -v
error "planettype not supported" now gets piped to output
-p now shows a table even if fifa resultset is empty
ski now checks in folder sql for sql script
ski now checks planettype and scripttype correctness

MrBlaTi committed:
fixed test, fixed version

MrBlaTi committed:
fixed tablecoloring, removed metainfos

Commit:
updated changelog

Commit:
updated changelog

MrBlaTi committed:
updated changelog

MrBlaTi committed:
changed size of titles

MrBlaTi committed:
fixed error

MrBlaTi committed:
merge #1

MrBlaTi committed:
merge #2

nbatsukh commented on src/IO.go:
```diff
@@ -21,9 +22,18 @@ func formatAndPrint(planets []Planet, opts *Opts, writer io.Writer) {
 	log.Debugf("using formatter of type : %T", formatter)
 	formatter.init()
 	formatter.format(planets, opts, writer)
+
```
redundant. It is done in ski.go in the method handleExitCode.

MrBlaTi replied:
fixed

nbatsukh commented on src/IO.go:
```diff
@@ -56,3 +66,14 @@ func makeDir(name string) {
+func makeRed(input string) string {
+	tokens := strings.Split(input, "\n")
+	for i, row := range tokens {
+		if row == "" {
```
"\t", "      " etc. not checked.

MrBlaTi replied:
No need to.
Format of the ascii table has been changed to never altering the cells formatting in favor of format sensitive output. This way No special attention is needed for inline structures as the whole line is colored.

One question tho: Is it practical to search for \r as well? according to e.g. http://stackoverflow.com/questions/1761051/difference-between-n-and-r
\r is, if anything, used on old systems and electromechanical teletype-like "terminals"

nbatsukh commented on src/InformationProcessing.go:
```diff
@@ -131,3 +156,15 @@ func validateSkiFormat(fifaString string) bool {
 }
+
+func getScriptPath(opts *Opts) string {
+	sql := strings.HasSuffix(opts.ScriptName, ".sql")
```
Make the check more reliable/case insensitive. Only one check is needed.

sql := strings.hasSuffix(strings.ToLower(opts.ScriptName), ".sql")

nbatsukh commented on src/Planet.go:
```diff
@@ -53,6 +61,23 @@ func (planet *Planet) executeLinux(opts *Opts) {
+	if opts.ScriptName == "" {
+		return true
+	}
+	sql := strings.HasSuffix(opts.ScriptName, ".sql")
```
same problem as in getScriptPath function. See above.

nbatsukh commented on src/PrettyFormatter.go:
```diff
@@ -56,16 +56,28 @@ func (prettyFormatter *PrettyFormatter) addKey(key string) {
+	prettyFormatter.addEntry("ID", id, completeTable)
+	prettyFormatter.addEntry("Name", name, completeTable)
 	prettyFormatter.addEntry("Address", address, completeTable)
-	prettyFormatter.addEntry("Type", planet.planetType, completeTable)
```
how about renaming the makeRed() method to color(text string, flag bool). If flag is set then a color is applied otherwise the text is returned as is. 
adding new variables is not necessary IMHO.

nbatsukh commented on src/PrettyTableFormatter.go:
```diff
@@ -146,11 +157,21 @@ func (prettyTableFormatter *PrettyTableFormatter) createSetForPlanet(json string
+		address = makeRed(address)
+		planetType = makeRed(planetType)
+	}
 	prettyTableFormatter.addEntry("Nr.", number, table)
```
See above the comment for createSetForPlanet

nbatsukh commented on src/TableFormatter.go:
```diff
@@ -79,10 +79,10 @@ func (tableFormatter *TableFormatter) executeTextFSM(planet Planet, opts *Opts)
-		full := fmt.Sprintf("%s\n --- Additional info: %s\n", err, message)
-		planet.outputStruct.output = fmt.Sprintf("%s\n%s", planet.outputStruct.output, full)
+		errorstring := fmt.Sprintf("%s\n --- Additional info: %s\n", err, message)
+		planet.outputStruct.output = fmt.Sprintf("%s\n%s\n", planet.outputStruct.output, makeRed(errorstring))
```
How about adding an error field to outputStruct on the planet for error messages. At the end of the execution you will have to mark only one chunk of text red instead of peppering the whole code base with calls to makeRed function.

nbatsukh commented on src/ski.go:
```diff
@@ -67,9 +68,9 @@ func isValidPlanet(planet Planet) bool {
 		switch planet.planetType {
 		case webServer:
-			fmt.Fprintf(os.Stderr, "Usage of ski with web servers is not implemented")
+			planet.outputStruct.output += makeRed("Usage of ski with web servers is not implemented\n")
```
How about adding an error field to outputStruct on the planet for error messages. At the end of the execution you will have to mark only one chunk of text red instead of peppering the whole code base with calls to makeRed function.

MrBlaTi committed:
fixes for merge
removed redundant code
renamed makeRed() to something more general
improved isBlank detection in coloring strings

MrBlaTi committed:
renamed "jobs_output" to "reports"

nbatsukh commented on src/IO.go:
```diff
@@ -67,13 +66,22 @@ func makeDir(name string) {
 	return strings.Join(tokens, "\n")
 }
+
+func isBlank(input string) bool {
```
there is a go function called strings.TrimSpace which does exactly what you are doing here.

strings.TrimSpace(" \t\n a lone gopher \n\t\r\n")

MrBlaTi committed:
removed manual trim space with existing one

added windows color support

fixed issues

fixed really weird mistake

wip

katzer commented on src/InformationProcessing.go:
```diff
@@ -57,14 +65,30 @@ func parseConnectionDetails(ids []string) []Planet {
-		} else {
-			keyPath = path.Join(os.Getenv("ORBIT_HOME"), "config", "ssh", "orbit.key")
-		}
+		keyPath = path.Join(os.Getenv("ORBIT_HOME"), "config", "ssh", "orbit.key")
```
`keyPath` bitte in $ORBIT_HOME/config/__keys__/orbit.key ändern.

katzer commented on src/InformationProcessing.go:
```diff
@@ -86,14 +110,17 @@ func getFullSkiString(ids []string) []string {
 	}
 
-	args := append([]string{"-f=ski", "--no-color"}, ids...)
+	args := append([]string{"-f=ski", "--no-colorize"}, ids...)
```
`--no-color` war schon richtig

katzer commented on src/VersionHandler.go:
```diff
@@ -10,32 +10,34 @@ import (
 	switch runtime.GOOS {
 	case windows:
-		return "Windows"
+		return "Windows NT"
```
`Windows_NT`

katzer commented on src/VersionHandler.go:
```diff
@@ -51,15 +53,7 @@ func getOSArch() string {
-			return "x86_64"
-		}
-		return "i686"
+		return "x86_64"
```
Für Windows findest du die Info in 

    $ echo %PROCESSOR_ARCHITECTURE%

bzw.

    $ [environment]::Is64BitOperatingSystem

katzer reviewed (commented):
Does the coloured output also works for cmd.exe and powershell?
